README.md

```diff
@@ -1,103 +1,66 @@
+  *.cpp        - Contains the bulk of the library that is used both runtime and by the compilers.
+                 Suggestion is to build src/*.cpp into a library.
 tool\
   dltlc\
-    dltlc.cpp - this is the source to the typelib-compiler, also called dltlc. Suggestion is to build this as an exe and link to above built lib.
+    dltlc.cpp  - This is the source to the typelib-compiler, also called dltlc. Suggestion
+                 is to build this as an exe and link to above built lib.
   dlpack\
-    dlpack.cpp - this is a tool that wraps the lib built from src/ to perform json -> binary, binary -> json and inspection of binary instances.
+    dlpack.cpp - This is a tool that wraps the lib built from src/ to perform json -> binary,
+                 binary -> json and inspection of binary instances.
 ```
 
-However if you want to build the library with bam, as is done during development and on autobuilders, these are the steps needed.
+## The Worklow of Data Library
 
-Run script/bootstrap.sh or script/bootstrap.bat depending on your platform, this will only have to be done once.
-	- these scripts will expect there to be a compiler and git in your PATH ( 'gcc' or 'clang' on 'unix', 'msvc' on windows )
-	- after this step there should be a local/bam/bam(.exe)
+1. Define Type Library definitions for your structures (.tlc files). tlc files are used by Data Library load/store types.
+2. Run dltlc (Data Library Type Library Compiler) to compile .tlc files. This step outputs all pieces necessary to load or store data at run-time.
+3. During run-time the generated header and functions from dl.h or dl.utils.h can be used to perform load/store operations.
 
-Run `local/bam/bam platform=<platform> config=<config> compiler=<compiler> -r sc`
-	- this will build all target for the specified platform and configuration into `local/<platform>/<compiler>/<config>`
+### The Various Parts of Data Library
 
-	- platforms:
-		* linux_x86    - 32bit linux x86 build.
-		* linux_x86_64 - 64bit linux x86 build.
-		* win32        - 32bit windows build.
-		* winx64       - 64bit windows build.
-
-	- configs
-		* debug        - non-optimized build
-		* release      - optimized build
-		* sanitizer    - sanitizer build
-		* coverage     - special config used for code-coverage on autobuilders
+* **load**: read an instance from packed form into memory
+* **store**: write an instance to packed form from memory
+* **pack**: pack txt/json instance to packed format
+* **unpack**: unpack from packed format to txt/json
+* generated C/C++ headers from dltlc
+* **type library (tlc)**: the file format used by Data Library itself to load/store/pack/unpack types.
+* **C-library**: data library itself, all the .cpp files in the src directory.
+* **dltlc**: data library type-library-compiler
+* **dl_pack**: tool to pack/unpack/convert instances to/from JSON.
+* **bindings**: python, lua
```

Comment:
About the bindings... I'm thinking about removing them, or at least move them to separate repos as they aren't really maintained.